src/module/helpers/sockets.mjs

```diff
@@ -28,4 +33,36 @@ export default class DrawSteelSocketHandler {
   emit(type, payload) {
     game.socket.emit(this.identifier, {type, payload});
   }
+
+  /**
+   * Tell the GM to spend a hero token
+   * @param {object} payload
+   * @param {string} payload.userId
+   * @param {string} payload.spendType
+   */
+  async spendHeroToken({userId, spendType}) {
```

Comment:
Do we expect to add a lot more socket events? I can imagine we will, for example for forced movement of an enemy. In which case, what did you think of my PR for a socket manager for the dnd5e system?

Reply:
Yes absolutely expect more socket events to get around the permission structure.

Oh, interesting work on enforcing certain kinds of data in the payloads by rebinding the entire structure. I also like tying together the emission & handling.